pkg/loader/translate_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loader
+
+import (
+	pb "github.com/pingcap/tidb-tools/tidb-binlog/slave_binlog_proto/go-binlog"
+
+	. "github.com/pingcap/check"
+)
+
+type slaveBinlogToTxnSuite struct{}
+
+var _ = Suite(&slaveBinlogToTxnSuite{})
+
+func (s *slaveBinlogToTxnSuite) TestTranslateDDL(c *C) {
+	db, table := "test", "hello"
+	sql := "CREATE TABLE hello (id INT AUTO_INCREMENT) PRIMARY KEY(id);"
+	binlog := pb.Binlog{
+		Type: pb.BinlogType_DDL,
+		DdlData: &pb.DDLData{
+			SchemaName: &db,
+			TableName:  &table,
+			DdlQuery:   []byte(sql),
+		},
+	}
+	txn := SlaveBinlogToTxn(&binlog)
+	c.Assert(txn.DDL.Database, Equals, db)
+	c.Assert(txn.DDL.Table, Equals, table)
+	c.Assert(txn.DDL.SQL, Equals, sql)
+}
+
+func (s *slaveBinlogToTxnSuite) TestTranslateDML(c *C) {
+	db, table := "test", "hello"
+	var oldVal, newVal int64 = 41, 42
+	dml := pb.DMLData{
+		Tables: []*pb.Table{
+			{
+				SchemaName: &db,
+				TableName:  &table,
+				ColumnInfo: []*pb.ColumnInfo{
+					{Name: "uid"},
+				},
+				Mutations: []*pb.TableMutation{
+					{
+						Type: pb.MutationType_Update.Enum(),
+						Row: &pb.Row{
+							Columns: []*pb.Column{
+								{Int64Value: &newVal},
+							},
+						},
+						ChangeRow: &pb.Row{
+							Columns: []*pb.Column{
+								{Int64Value: &oldVal},
+							},
+						},
+					},
+					{
+						Type: pb.MutationType_Insert.Enum(),
```

Comment:
add case for `DeleteDMLType` and `UnknownDMLType` too?

Reply:
If we want, we can add those two in tests for `getDMLType`.